test/sharding.go

```diff
@@ -121,17 +121,22 @@ func (sth *ShardingTestHelper) GetTreeSerialFile(t *testing.T) files.Directory {
 // a testing random file of the given size (in kbs). The random
 // file is different every time.
 func (sth *ShardingTestHelper) GetRandFileMultiReader(t *testing.T, kbs int) (*files.MultiFileReader, io.Closer) {
-	st := sth.makeRandFile(t, kbs)
-	sf, err := files.NewSerialFile(sth.path(shardingTestFile), false, st)
+	slf, sf := sth.GetRandFileReader(t, kbs)
```

Comment:
This is completely changing the semantics of these methods. Before they were readers to a plain file, now they reading files from a directory. This should not return a MultiFileReader with an underlying SliceFile. Could you explain this change?

Reply:
The old version is returning a directory too (`files.File` was used for both files and dirs) - `NewSliceFile` - with one regular file - `NewSerialFile` called `randomfile`.

One change here is that `NewMultiFileReader` can now return an error (which is actually not needed anymore, I'll change this in `go-ipfs-file`)